### ISSUE TITLE
Predicate and limit pushdown for system tables

### ISSUE THREAD

**Issue by tustvold** [influxdata/influxdb_iox#2477]
Currently predicates and limits are not pushed down into [scans for system tables](https://github.com/influxdata/influxdb_iox/blob/main/server/src/db/system_tables.rs#L139), this combined with the fact the current design aims to produce a [single RecordBatch](https://github.com/influxdata/influxdb_iox/blob/main/server/src/db/system_tables.rs#L116) results in query timeouts on systems with large catalogs.

This should be fixed.

For reference the query I was running is

```
select table_name,
       partition_key,
       row_count,
       (cast(min_value as bigint) - cast(previous_max_value as bigint)) / 1000000000 as delta_seconds
from (select *,
             LAG(max_value)
             OVER (partition by table_name, partition_key order by table_name, partition_key, chunk_id) as previous_max_value
      from system.chunk_columns
      where column_name = 'time'
        and storage = 'ReadBufferAndObjectStore' and table_name like 'http_api%');
```

**Comment by alamb** [comment]
This is a valuable feature / idea for sure. 👍

**Comment by alamb** [comment]
TLDR: as implied by this ticket's description, the vast majority of time spent in system table queries is spent creating the `RecordBatch`  for the queries. Adding support for `limit`, `projection` and `predicates` will help.

## Analysis
I inserted a "bunch" of data into my local instance using load generator with commands like

```shell
../target/release/iox_data_generator -h localhost:8080 --token foo --org org --bucket bucket -s schemas/cap-write.toml  --start '356 days ago' --batch_size=1000
```

This resulted in  `1122347` rows in the `system.column_chunks` table and I ran the queries (with a release `influxdb_iox` build)  as shown under "Query Timings".

I observed a few things:
1. The complexity of the query did not appreciably affect query performance (`count(*)` took about the same as something with several subqueries) ;
1. Creating a `String` for each UUID (which are then copied and freed) was a hot spot in the profile (see screen shot below)
3. There is no `limit` in theses queries but I often write queries like `select * from system.chunks limit 10` as a way to see the schema of the table.


## Unscientific Instruments profile
![Screen Shot 2021-11-17 at 3 00 00 PM](https://user-images.githubusercontent.com/490673/142288180-b6575875-923d-4941-ba67-0c7541a959c9.png)

## Thoughts and Analysis

There are many different ways to optimize system table creation so one
key consideration is how to avoid spending too much effort here.

Evaluating predicates on arbitrary columns in IOx code is not likely
to be all that much more efficient than using the generic DataFusion
executor, but will be non trivially complex so we should likely only pursue that as a last resort.

The `chunks`, `operations` and `persistence` tables are already created column by column (the most efficient way to create arrow arrays). The `chunk_summaries` table is not quite as straightforward as it is formed row by row, but I think we can change that in a straightforward manner

Another observation is that some queries (like `count(*)` do not need
to read many columns at all (the required columns are pushed down as a
`projection` by DataFusion) so we can skip columns that do not appear
in the query


## Plan

Thus, with the above analysis, my proposed plan is, in order, measuring performance as I got and stopping when the improvement doesn't seem that large.

- [ ] switch `chunk_columns` to be created column wise
- [ ] Implement `projection` pushdown for on system tables
- [ ] Implement `limit` pushdown for system tables
- [ ] (Stretch goal) Implement limited `filter` pushdown for the `table_name` for `chunks` and `persistence`, and `table_name` and `column_name` for `column_chunks `(based on the predicate pruning logic in DataFusion)

Bonus points for
- [ ] File a ticket to improve `StringArray` creation in Arrow to things that `impl Display` rather than `&str` (so we could form strings directly from `UUIDs` into the right location rather than reallocating storage over and over again)
